### PR TITLE
fix async destination order

### LINKED ISSUE #28154: [Async Destinations] -flush worker error: java.lang.NullPointerException
@rodireich  is attempting to sync 1TB to destination snowflake.  We see it crashing eventually around flush operations

https://cloud.airbyte.com/workspaces/23ef6ec3-cbc7-460f-bbfb-2b88a7716c53/connections/951e1359-f56e-4a3e-a161-9b511d0efd20/job-history#3086715::2

```
2023-07-11 01:38:20 destination > ERROR i.a.i.d.FlushWorkers(lambda$flush$1):168 Flush Worker (69f89) -- flush worker error: java.lang.NullPointerException: Cannot invoke "Object.hashCode()" because "key" is null
        at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936) ~[?:?]
        at io.airbyte.integrations.destination_async.state.GlobalAsyncStateManager.flushStates(GlobalAsyncStateManager.java:153) ~[io.airbyte.airbyte-integrations.bases-base-java-24.0.2.jar:?]
        at io.airbyte.integrations.destination_async.buffers.MemoryAwareMessageBatch.flushStates(MemoryAwareMessageBatch.java:69) ~[io.airbyte.airbyte-integrations.bases-base-java-24.0.2.jar:?]
        at io.airbyte.integrations.destination_async.FlushWorkers.lambda$flush$1(FlushWorkers.java:161) ~[io.airbyte.airbyte-integrations.bases-base-java-24.0.2.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

### PR DESCRIPTION
**Pull request by cgardens**
## What
Partly closes https://github.com/airbytehq/oncall/issues/2437 - a rare condition that can occur when comparator values change mid-comparison.

Closes https://github.com/airbytehq/airbyte/issues/28154 - a NPE when retrieving from the queue.

## How
- For the comparator error, we are unfortunately unable to write a test case to prove the error. It's exceedingly difficult to force the `java.lang.IllegalArgumentException: Comparison method violates its general contract!` error. We do know the current implementation contains an error this PR fixes.
- The NPE error is straight-forward. The peek returns a null and we were not accounting for that before.

## Recommended reading order
Both files.

### CODE REVIEW

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/DetectStreamToFlush.java [hunk 1]
```diff
@@ -210,11 +211,18 @@ long estimateSizeOfRunningWorkers(final StreamDescriptor stream, final long curr
...
   List<StreamDescriptor> orderStreamsByPriority(final Set<StreamDescriptor> streams) {
+    // eagerly pull attributes so that values are consistent throughout comparison
+    final Map<StreamDescriptor, Optional<Long>> sdToQueueSize = streams.stream()
+        .collect(Collectors.toMap(s -> s, bufferDequeue::getQueueSizeBytes));
+
+    final Map<StreamDescriptor, Optional<Instant>> sdToTimeOfLastRecord = streams.stream()
+        .collect(Collectors.toMap(s -> s, bufferDequeue::getTimeOfLastRecord));
+
     return streams.stream()
-        .sorted(Comparator.comparing((StreamDescriptor s) -> bufferDequeue.getQueueSizeBytes(s).orElseThrow(), Comparator.reverseOrder())
```
Comment (davinchia) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/DetectStreamToFlush.java:214]: Our theory is prior to this implementation, these values could change mid-comparison, and thus cause the `java.lang.IllegalArgumentException: Comparison method violates its general contract!` we see.

**Comment by octavia-squidington-iii** [comment]
##  source-snowflake test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ❌

⏲️  Total pipeline duration: 09mn42s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                              | ✅        |
| QA checks                                                                   | ✅        |
| Build connector tar                                                         | ✅        |
| Build source-snowflake docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-snowflake:integrationTest | ❌        |
| Acceptance tests                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/source-snowflake/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-scylla test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ❌

⏲️  Total pipeline duration: 05mn30s 

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-scylla/metadata.yaml     | ✅        |
| Connector version semver check                                                | ✅        |
| QA checks                                                                     | ✅        |
| Build connector tar                                                           | ✅        |
| Build destination-scylla docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-scylla:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-scylla/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-scylla test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-databricks test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ✅

⏲️  Total pipeline duration: 16mn25s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-databricks/metadata.yaml     | ✅        |
| Connector version semver check                                                    | ✅        |
| QA checks                                                                         | ✅        |
| Build connector tar                                                               | ✅        |
| Build destination-databricks docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-databricks:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-databricks/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-databricks test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-mssql-strict-encrypt test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ❌

⏲️  Total pipeline duration: 10mn05s 

| Step                                                                                        | Result   |
|:--------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-mssql-strict-encrypt/metadata.yaml     | ✅        |
| Connector version semver check                                                              | ✅        |
| QA checks                                                                                   | ✅        |
| Build connector tar                                                                         | ✅        |
| Build destination-mssql-strict-encrypt docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-mssql:dev                                                       | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-mssql-strict-encrypt:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-mssql-strict-encrypt/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-mssql-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-kinesis test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ✅

⏲️  Total pipeline duration: 09mn35s 

| Step                                                                           | Result   |
|:-------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-kinesis/metadata.yaml     | ✅        |
| Connector version semver check                                                 | ✅        |
| QA checks                                                                      | ✅        |
| Build connector tar                                                            | ✅        |
| Build destination-kinesis docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-kinesis:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-kinesis/0.1.5/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-kinesis test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-mongodb test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ❌

⏲️  Total pipeline duration: 16mn31s 

| Step                                                                           | Result   |
|:-------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-mongodb/metadata.yaml     | ✅        |
| Connector version semver check                                                 | ✅        |
| QA checks                                                                      | ✅        |
| Build connector tar                                                            | ✅        |
| Build destination-mongodb docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-mongodb:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-mongodb/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-mongodb test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-mysql-strict-encrypt test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ❌

⏲️  Total pipeline duration: 29mn45s 

| Step                                                                                        | Result   |
|:--------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-mysql-strict-encrypt/metadata.yaml     | ✅        |
| Connector version semver check                                                              | ✅        |
| QA checks                                                                                   | ✅        |
| Build connector tar                                                                         | ✅        |
| Build destination-mysql-strict-encrypt docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-mysql:dev                                                       | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-mysql-strict-encrypt:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-mysql-strict-encrypt/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-mysql-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-cassandra test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ❌

⏲️  Total pipeline duration: 17.39s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-cassandra/metadata.yaml | ✅        |
| Connector version semver check                                               | ✅        |
| QA checks                                                                    | ✅        |
| Build connector tar                                                          | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-cassandra/0.1.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-cassandra test
```

**Comment by octavia-squidington-iii** [comment]
##  source-azure-blob-storage test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ✅

⏲️  Total pipeline duration: 10mn37s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/source-azure-blob-storage/0.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-azure-blob-storage test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ✅

⏲️  Total pipeline duration: 17mn32s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-bigquery-denormalized/1.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ✅

⏲️  Total pipeline duration: 16.42s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-bigquery/1.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-teradata test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ❌

⏲️  Total pipeline duration: 47mn41s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-teradata/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-teradata docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-teradata:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-teradata/0.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-teradata test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-e2e-test test report (commit [`268bfdf53f`](https://github.com/airbytehq/airbyte/pull/28257/commits/268bfdf53fb8a63687505865ca73356ad5fad4ea)) - ❌

⏲️  Total pipeline duration: 05mn00s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-e2e-test/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-e2e-test docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-e2e-test:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689227371/268bfdf53fb8a63687505865ca73356ad5fad4ea/destination-e2e-test/0.3.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-e2e-test test
```

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/DetectStreamToFlush.java (hunk 1 above)
Comment (edgao) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/DetectStreamToFlush.java:214]: this theory makes sense to me! fun times with 🧵🧵🧵

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/state/GlobalAsyncStateManager.java [hunk 2]
```diff
@@ -150,8 +150,12 @@ public List<PartialAirbyteMessage> flushStates() {
       while (true) {
         final Long oldestState = stateIdQueue.peek();
         final boolean emptyQ = oldestState == null;
+        if (emptyQ) {
```
Comment (edgao) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/state/GlobalAsyncStateManager.java:153]: nit: might be a bit more readable as
```java
if (stateIdQueue.isEmpty()) {
  break;
} 
final Long oldestState = stateIdQueue.peek();
```
(iirc that's the only time peek returns null)

**Review by edgao — APPROVED** [review]

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/state/GlobalAsyncStateManager.java (hunk 2 above)
Comment (davinchia) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/state/GlobalAsyncStateManager.java:153]: I like it! Now I have to run the tests..

**Comment by davinchia** [comment]
I'm not sure why the CI isn't building this:
- the legacy test command for snowflake is passing
- the base-java tests are passing for me locally


```
SUB_BUILD=CONNECTORS_BASE ./gradlew :airbyte-integrations:bases:base-java:test
...
IntegrationRunnerTest > testDiscover() PASSED

IntegrationRunnerTest > testSpecDestination() PASSED

IntegrationRunnerTest > testDestinationConsumerLifecycleFailure() PASSED

IntegrationRunnerTest > testReadException() PASSED

IntegrationRunnerTest > testSpecSource() PASSED

IntegrationRunnerTest > testInterruptOrphanThreadFailure() PASSED

SerializedBufferingStrategyTest > testPerStreamThresholdFlush() PASSED

SerializedBufferingStrategyTest > testConcurrentStreamThresholdFlush() PASSED

SerializedBufferingStrategyTest > testCreateBufferFailure() PASSED

SerializedBufferingStrategyTest > testTotalStreamThresholdFlush() PASSED

RunningFlushWorkersTest > testCompleteFlushWorkerWithoutTrackThrowsException() PASSED

RunningFlushWorkersTest > testCompleteFlushWorker() PASSED

RunningFlushWorkersTest > testMultipleStreams() PASSED

RunningFlushWorkersTest > testTrackFlushWorker() PASSED

RunningFlushWorkersTest > testGetSizesOfRunningWorkerBatches() PASSED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 39s
```

**Comment by octavia-squidington-iii** [comment]
##  source-teradata test report (commit [`4ee1ca812e`](https://github.com/airbytehq/airbyte/pull/28257/commits/4ee1ca812e807f81234ccae2d1d81c4788b0eab6)) - ✅

⏲️  Total pipeline duration: 12mn25s 

| Step                                                                       | Result   |
|:---------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-teradata/metadata.yaml     | ✅        |
| Connector version semver check                                             | ✅        |
| QA checks                                                                  | ✅        |
| Build connector tar                                                        | ✅        |
| Build source-teradata docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-teradata:integrationTest | ✅        |
| Acceptance tests                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689313778/4ee1ca812e807f81234ccae2d1d81c4788b0eab6/source-teradata/0.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-teradata test
```

**Comment by octavia-squidington-iii** [comment]
##  source-db2 test report (commit [`4ee1ca812e`](https://github.com/airbytehq/airbyte/pull/28257/commits/4ee1ca812e807f81234ccae2d1d81c4788b0eab6)) - ❌

⏲️  Total pipeline duration: 05mn36s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-db2/metadata.yaml     | ✅        |
| Connector version semver check                                        | ✅        |
| QA checks                                                             | ✅        |
| Build connector tar                                                   | ✅        |
| Build source-db2 docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-db2:integrationTest | ❌        |
| Acceptance tests                                                      | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689313778/4ee1ca812e807f81234ccae2d1d81c4788b0eab6/source-db2/0.1.20/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-db2 test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-mysql test report (commit [`4ee1ca812e`](https://github.com/airbytehq/airbyte/pull/28257/commits/4ee1ca812e807f81234ccae2d1d81c4788b0eab6)) - ❌

⏲️  Total pipeline duration: 39mn40s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-mysql/metadata.yaml     | ✅        |
| Connector version semver check                                               | ✅        |
| QA checks                                                                    | ✅        |
| Build connector tar                                                          | ✅        |
| Build destination-mysql docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-mysql:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-mysql:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689313778/4ee1ca812e807f81234ccae2d1d81c4788b0eab6/destination-mysql/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-mysql test
```

**Comment by octavia-squidington-iii** [comment]
##  source-bigquery test report (commit [`4ee1ca812e`](https://github.com/airbytehq/airbyte/pull/28257/commits/4ee1ca812e807f81234ccae2d1d81c4788b0eab6)) - ✅

⏲️  Total pipeline duration: 13mn22s 

| Step                                                                       | Result   |
|:---------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                             | ✅        |
| QA checks                                                                  | ✅        |
| Build connector tar                                                        | ✅        |
| Build source-bigquery docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-bigquery:integrationTest | ✅        |
| Acceptance tests                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689313778/4ee1ca812e807f81234ccae2d1d81c4788b0eab6/source-bigquery/0.3.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-pubsub test report (commit [`4ee1ca812e`](https://github.com/airbytehq/airbyte/pull/28257/commits/4ee1ca812e807f81234ccae2d1d81c4788b0eab6)) - ✅

⏲️  Total pipeline duration: 14mn50s 

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-pubsub/metadata.yaml     | ✅        |
| Connector version semver check                                                | ✅        |
| QA checks                                                                     | ✅        |
| Build connector tar                                                           | ✅        |
| Build destination-pubsub docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-pubsub:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/cgardens_fix-async-destination-order/1689313778/4ee1ca812e807f81234ccae2d1d81c4788b0eab6/destination-pubsub/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-pubsub test
```

**Pull request merged**